### PR TITLE
Merge master into Release v0.4.0

### PR DESCRIPTION
**Pull request by RainFallsSilent**

### CODE REVIEW

File: blockchain/chaindata.go [hunk 1]
```diff
@@ -254,7 +254,7 @@ func (c *ChainStore) RollbackUnspendUTXOs(b *Block) error {
 				var err error
 				unspendUTXOs[programHash][assetID][height], err = c.GetUnspentElementFromProgramHash(programHash, assetID, height)
 				if err != nil {
-					return errors.New(fmt.Sprintf("[persist] UTXOs programHash:%v, assetID:%v has no unspent UTXO.", programHash, assetID))
+					fmt.Println(fmt.Sprintf("[persist] UTXOs programHash:%v, assetID:%v has no unspent UTXO.", programHash, assetID))
```
Comment (Houshoupei84) [inline blockchain/chaindata.go]: may be log is more proper

File: dpos/manager/proposaldispatcher.go [hunk 2]
```diff
@@ -463,6 +463,7 @@ func (p *ProposalDispatcher) IsCRCBadNetWork() bool {
 }
 
 func (p *ProposalDispatcher) IsViewChangedTimeOut() bool {
+	return false
```
Comment (Houshoupei84) [inline dpos/manager/proposaldispatcher.go]: is return false Temporary? If so, maybe need add todo and If not del no use code .

Comment (RainFallsSilent) [reply dpos/manager/proposaldispatcher.go]: This change has been recorded in trello. It is a temp change to fix bug of main net.

File: blockchain/chaindata.go (hunk 1 above)
Comment (RainFallsSilent) [reply blockchain/chaindata.go]: fixed

**Pull request merged**